File: app/src/main/java/com/qimian233/ztool/hook/modules/systemui/qs/ControlCenterNoTileLabelsHook.java
package com.qimian233.ztool.hook.modules.systemui.qs;

import android.annotation.SuppressLint;
import android.view.View;
import android.view.ViewGroup;

import com.qimian233.ztool.hook.base.AppHookModule;

import io.github.libxposed.api.XposedModuleInterface;

import java.lang.reflect.Method;

@SuppressLint("PrivateApi")
public class ControlCenterNoTileLabelsHook extends AppHookModule {

    public ControlCenterNoTileLabelsHook() {}

    @Override
    public String getModuleName() {
        return "control_center_no_tile_labels";
    }

    @Override
    public String[] getTargetPackages() {
        return new String[]{"com.android.systemui"};
    }

    @Override
    public void handleLoadPackage(XposedModuleInterface.PackageLoadedParam param) throws Throwable {
        ClassLoader classLoader = param.getDefaultClassLoader();
        Method createAndAddLabelsMethod = classLoader
                .loadClass("com.android.systemui.qs.tileimpl.CustomQSTileViewImpl")
                .getDeclaredMethod("createAndAddLabels");
        hookWithId(createAndAddLabelsMethod, "create_and_add_labels", chain -> {
            Object result = chain.proceed();
            try {
                Class<?> cl = chain.getThisObject().getClass();
                ViewGroup labelContainer = (ViewGroup) findField(cl,  "labelContainer")
                        .get(chain.getThisObject());
                if (labelContainer != null) {
                    labelContainer.setVisibility(View.GONE);
                    labelContainer.setImportantForAccessibility(View.IMPORTANT_FOR_ACCESSIBILITY_NO_HIDE_DESCENDANTS);
                }
            } catch (Exception e) {
                logger.error("Cannot apply no-label mode to tiles!", e);
            }
            return result;
        });
    }
}

File: app/src/main/java/com/qimian233/ztool/hook/modules/systemui/qs/ControlCenterNoTileLabelsHook.kt
package com.qimian233.ztool.hook.modules.systemui.qs

import android.annotation.SuppressLint
import android.view.View
import android.view.ViewGroup
import com.qimian233.ztool.data.keys.PreferenceKeys
import com.qimian233.ztool.data.keys.ScopeKeys
import com.qimian233.ztool.hook.base.AppHookModule
import io.github.libxposed.api.XposedModuleInterface.PackageLoadedParam
import java.lang.reflect.Method

@SuppressLint("PrivateApi")
class ControlCenterNoTileLabelsHook : AppHookModule() {

    override fun getModuleName(): String = PreferenceKeys.CONTROL_CENTER_NO_TILE_LABELS.name

    override fun getTargetPackages(): Array<String> = arrayOf(ScopeKeys.SYSTEM_UI.packageName)

    override fun handleLoadPackage(param: PackageLoadedParam) {
        val classLoader = param.defaultClassLoader
        val createAndAddLabelsMethod: Method = classLoader
            .loadClass("com.android.systemui.qs.tileimpl.CustomQSTileViewImpl")
            .getDeclaredMethod("createAndAddLabels")
        hookWithId(createAndAddLabelsMethod, "create_and_add_labels") { chain ->
            val result = chain.proceed()
            try {
                val cl = chain.thisObject.javaClass
                val labelContainer = findField(cl, "labelContainer")
                    .get(chain.thisObject) as? ViewGroup
                if (labelContainer != null) {
                    labelContainer.visibility = View.GONE
                    labelContainer.importantForAccessibility =
                        View.IMPORTANT_FOR_ACCESSIBILITY_NO_HIDE_DESCENDANTS
                }
            } catch (e: Exception) {
                logger.error("Cannot apply no-label mode to tiles!", e)
            }
            result
        }
    }
}
